src/libraries/System.Private.CoreLib/src/System/Threading/AutoreleasePool.cs

```diff
@@ -8,21 +8,20 @@ namespace System.Threading
 {
     internal static class AutoreleasePool
     {
-        private static bool CheckEnableAutoreleasePool()
+        private static int s_enableAutoreleasePool;
+        public static bool EnableAutoreleasePool
         {
-            const string feature = "System.Threading.Thread.EnableAutoreleasePool";
-#if !CORECLR
-            return AppContextConfigHelper.GetBooleanConfig(feature, false);
-#else
-            bool isEnabled = CLRConfig.GetBoolValue(feature, out bool isSet);
```

Comment:
This was done this way because the configuration value for this is needed prior to `AppContextConfigHelper` being populated. The QCall for `GetBoolValue` pokes into the native data structure that has the values that eventually populate `AppContextConfigHelper`.

Reply:
@am11 I just saw your comments above. The tests should fail if this is needed.

Reply:
Do we have tests in Pri0 category/? All tests were passing on Windows x64 Pri0 in the previous run.

Reply:
The specific line I am thinking of that may need this is the `Start()` on host. That method does a lot and it isn't obvious but this logic is done early in thread initialization.

https://github.com/dotnet/runtime/blob/4017327955f1d8ddc43980eb1848c52fbb131dfc/src/coreclr/dlls/mscoree/unixinterface.cpp#L251-L254

Reply:
This is for [Autoreleasepool](https://developer.apple.com/documentation/foundation/nsautoreleasepool) on macOS for Objective-C.

I did just take a quick look at where we are doing this and am less concerned now. I think this will be okay.

> Do we have tests in Pri0 category/?

Yes. But as mentioned above, this is a macOS only feature.

Reply:
@am11 sigh... I was about to go to bed and then the issue hit me. This isn't going work but our tests will likely pass. Let me explain.

The initialization of the AutoreleasePools on threads has 3 separate locations.

1) The execution of the main entry point. Found in `Assembly::ExecuteMainMethod()` and which is only called from `coreclr_execute_assembly`.

2) The managed code that runs at the early lifetime of a managed `Thread`, found in `Thread.RunWorker()`.

3) The initialization of the Finalizer thread, so that it can drain AutoreleasePools for Objective-C interop. This is similar to the clean-up that RCWs go through during finalization with the built-in COM interop.

The code in `coreclr_initialize` is all run prior to (1) and (2). However, for (3) the place I pointed out in `unixinterface.cpp`, line 251, is the call that eventually launches the Finalizer thread.

With this change there is now a race between the call to `CreateAppDomainWithManager` in `coreclr_initialize` and the Finalizer thread starting, that will call into the changed managed code. It is entirely possible the AppContext update has happened, but I'd prefer not to tempt fate in this case.

The test issue is going to also be difficult but now we need to validate the Finalizer thread has an AutoreleasePool installed. Confirming this is relatively hard and will likely not be reliable given the difficulties had with testing initially&mdash;the Objective-C runtime has a lot of "helpful" fallbacks and none of them are documented.

I'd prefer if you revert your last commit and update/add a comment in the `AutoreleasePool.cs` file to indicate this so if it becomes a real concern, someone who isn't me will see it and properly audit the trouble path.

Reply:
Thank you for the details, @AaronRobinsonMSFT. I have captured the reasoning in a comment and reverted the CLRConfig.cs deletion.